db/sql/sql.go

```diff
@@ -705,6 +703,30 @@ func (tx *Transaction) List(s *schema.Schema, filter transaction.Filter, pg *pag
 	return
 }
 
+//List resources in the db
+func (tx *Transaction) List(s *schema.Schema, filter transaction.Filter, pg *pagination.Paginator) (list []*schema.Resource, total uint64, err error) {
+	sql, args, err := buildSelect(s, filter, pg)
+	if err != nil {
+		return nil, 0, err
+	}
+
+	return executeSelect(s, filter, sql, args, tx)
+}
+
+//Lock resources in the db
+func (tx *Transaction) LockList(s *schema.Schema, filter transaction.Filter, pg *pagination.Paginator) (list []*schema.Resource, total uint64, err error) {
```

Comment:
I agree with adding special handling of related resources.

On the other hand, it's worth to mention that when using pagination we may lock more than expected ([gap locks](http://stackoverflow.com/a/5704927)). Example:
session1> set session transaction isolation level serializable;
session1> start transaction;
session1> select id, name from some_table order by id limit 1 offset 1 for update;
session2> set session transaction isolation level serializable;
session2> start transaction;
session2> select id, name from some_table order by id limit 1 for update;
session2> ERROR 1205 (HY000): Lock wait timeout exceeded; try restarting transaction

What's more, Fetch() uses List() and [filters results itself](https://github.com/cloudwan/gohan/blob/master/db/sql/sql.go#L773), so that's another situation of locking more than needed.

As a result, I don't think we could completely eliminate excessive locking.